src/Orchard.Web/Modules/Orchard.Users/Module.cs

```diff
@@ -60,13 +62,19 @@ public override void ConfigureServices(IServiceCollection serviceCollection)
 
             serviceCollection.TryAddScoped<IUserStore<User>, UserStore>();
 
+            var dataProtectionProvider = DataProtectionProvider.Create(_tenantName);
```

Comment:
The `DataProtectionProvider` static class was introduced to support non-DI environments (which allows using the new DP stack in OWIN/Katana apps for instance). Here, you should avoid using it, as devs won't be able to customize the DP options at the global level (e.g the folder where the key ring is stored).

Instead, consider resolving `IDataProtectionProvider` from the global DI container and create an `IDataProtector` by calling `CreateProtector(_tenantName)`. Then use this `IDataProtector` as your new `IDataProtectionProvider`.

You can apply the same transformation to the DP provider used by OpenIddict.